rosetta-lang/src/main/java/com/regnosys/rosetta/formatting2/RuneFormatter.java

```diff
@@ -0,0 +1,82 @@
+package com.regnosys.rosetta.formatting2;
+
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+
+import org.eclipse.emf.ecore.EObject;
+import org.eclipse.emf.ecore.resource.Resource;
+import org.eclipse.xtext.formatting2.FormatterRequest;
+import org.eclipse.xtext.formatting2.regionaccess.ITextRegionAccess;
+import org.eclipse.xtext.formatting2.regionaccess.ITextReplacement;
+import org.eclipse.xtext.formatting2.regionaccess.TextRegionAccessBuilder;
+import org.eclipse.xtext.formatting2.regionaccess.internal.TextRegionRewriter;
+import org.eclipse.xtext.nodemodel.INode;
+import org.eclipse.xtext.nodemodel.util.NodeModelUtils;
+import org.eclipse.xtext.resource.XtextResource;
+import org.eclipse.xtext.util.TextRegion;
+
+import javax.inject.Inject;
+import javax.inject.Provider;
+
+public class RuneFormatter implements CodeFormatterService{
+	@Inject
+	private Provider<FormatterRequest> formatterRequestProvider;
+	
+	@Inject 
+	private Provider<RosettaFormatter> rosettaFormatterProvider;
```

Comment:
This one is injecting the implementation rather than the interface. Change to inject the formatter interface instead.

Just to explain why: this is against the principle of dependency injection, since it breaks the separation between interface and implementation, making the codebase less extensible.